### ISSUE TITLE
flux kvs get/put need type-hint options

### ISSUE THREAD

**Issue by garlick** [flux-framework/flux-core#1159]
The `flux kvs get` command presumes that all values are JSON.  That assumption no longer holds after PR #1154.
```
Usage: flux-kvs get key [key...]
Get value stored under key
  -h, --help             Display this message.
```
We could add `--raw` and `--json` options.  If neither is specified, first attempt to decode as JSON, and if that fails, presume raw.

Also, supporting multiple keys on the command line doesn't really work now, since values are separated by a newline, and values can _contain_ a newline.   With raw data, this becomes even less useful, so we should probably not allow it.

The `flux kvs put` command assumes values are JSON, with a fallback to convert to a JSON string if the value doesn't decode as valid JSON.
```
Usage: flux-kvs put key=value [key=value...]
Store value under key
  -h, --help             Display this message.
```
Here again, let's support only a single key, and add `--json` and `--raw` options.  To support redirecting raw data from stdin, if `=value` is missing, stdin should be used. The fallback JSON string behavior should be eliminated.

**Comment by garlick** [comment]
Actually, I wonder if we should just have a `--json` option and presume raw if not specified?  That reflects the way the KVS actually works after #1158.  Also I should have noted that we'll need the raw txn/lookup functions that we decided to defer to another PR after #1158.

**Comment by grondo** [comment]
> Actually, I wonder if we should just have a --json option and presume raw if not specified?

I like this approach, though it is a change from how the command has always worked. However, long term it seems like the most reasonable UI for the command.
